### PR TITLE
Reduce subgraph requests

### PR DESCRIPTION
**Pull request by MohammadPCh**
- #4375 

## Summary by CodeRabbit

- **New Features**
	- Enhanced data fetching capabilities across multiple components using the `useQuery` hook from `@tanstack/react-query`.
	- Integration of user account information to allow dynamic interaction with subgraphs based on the user's blockchain address.
	- Introduced a new configuration property, `CHAINS_WITH_SUBGRAPH`, to dynamically manage chains with associated subgraph addresses.
	- Implemented periodic updates for state management in the `useRegenTokenDistroHelper` hook, ensuring data remains in sync.

- **Bug Fixes**
	- Removed unused commented-out code to improve readability and maintainability across multiple components.

- **Chores**
	- Updated import paths for types and configuration settings to streamline project structure and maintainability.
	- Increased the `SUBGRAPH_POLLING_INTERVAL` from `5000` to `300_000` for improved performance.

### CODE REVIEW

**Review by kkatusic — APPROVED** [review]
Great job, thx @MohammadPCh

**Review by Meriem-BM — APPROVED** [review]
LGTM!

File: src/components/GIVeconomyPages/GIVbacks.tsx [hunk 1]
```diff
@@ -43,27 +44,28 @@ import { GIVBackExplainModal } from '../modals/GIVBackExplain';
...
-	const chainId = chain?.id;
-	const values = useAppSelector(
-		state =>
-			chainId === config.OPTIMISM_NETWORK_NUMBER
-				? state.subgraph.optimismValues
-				: state.subgraph.gnosisValues,
-		() => (showHarvestModal ? true : false),
-	);
-
+	const { chain, address } = useAccount();
```
[the remark this answers is not in the document]
Comment (aminlatifi) [reply src/components/GIVeconomyPages/GIVbacks.tsx:56]: @MohammadPCh  I assume you need it for a consistent query key used in useQuery, right?

**Review by aminlatifi — CHANGES_REQUESTED** [review]
@MohammadPCh As we discussed, please implement liveness in at least components that need time-sensitive data like stake cards, stream reward, etc.
Thanks for this great job and migrating to react-query

**Comment by MohammadPCh** [comment]
@aminlatifi now the components will recalculate every 5 secs.

**Review by aminlatifi — APPROVED** [review]
Thanks @MohammadPCh

**Pull request merged**